packages/shared/src/components/tags/TagSelection.tsx

```diff
@@ -102,6 +104,15 @@ export function FilterOnboardingV4({
       select: tagsSelector,
     },
   );
+
+  const onboardingTags = useMemo(() => {
+    if (!shouldShuffleTags) {
+      return onboardingTagsRaw;
+    }
+
+    return onboardingTagsRaw?.sort(() => Math.random() - 0.5);
+  }, [shouldShuffleTags, onboardingTagsRaw]);
```

Comment:
One thing that comes to mind if this is SSR-ed it could point to server client mismatch because it randomises once on the server and once on the client. But I don't think its rendered on server side?